File: c#/UnitTestIntro/UnitTestsIntro/Calculator.cs
namespace UnitTestsIntro;

public static class Calculator
{
    public const string Add = "add";
    public const string Multiply = "multiply";
    public const string Divide = "divide";
    public const string Subtract = "subtract";

    private static readonly Dictionary<string, Func<int, int, int>> _supportedOperators =
        new()
        {
            {Add, (a, b) => a + b},
            {Multiply, (a, b) => a * b},
            {Divide, (a, b) => a / b},
            {Subtract, (a, b) => a - b}
        };
    public static int Calculate(int a, int b, string @operator) =>
        _supportedOperators.ContainsKey(@operator)
            ? _supportedOperators[@operator](a, b)
            : throw new ArgumentException("Not supported operator");
}

File: java/src/main/java/unit/tests/intro/Calculator.java
package unit.tests.intro;

import java.util.Map;
import java.util.function.BiFunction;

public class Calculator {
    public static final String ADD = "add";
    public static final String MULTIPLY = "multiply";
    public static final String DIVIDE = "divide";
    public static final String SUBTRACT = "subtract";

    private static final Map<String, BiFunction<Integer, Integer, Integer>> supportedOperators =
            Map.of(
                    ADD, (a, b) -> a + b,
                    MULTIPLY, (a, b) -> a * b,
                    DIVIDE, (a, b) -> a / b,
                    SUBTRACT, (a, b) -> a - b
            );

    public int calculate(int a, int b, String operator) throws IllegalArgumentException {
        if (!supportedOperators.containsKey(operator)) {
            throw new IllegalArgumentException("Not supported operator");
        }
        return supportedOperators.get(operator).apply(a, b);
    }
}